examples/gno.land/p/demo/grc/grc721/basic_nft.gno

```diff
@@ -353,12 +384,16 @@ func (s *basicNFT) mint(to std.Address, tid TokenID) error {
 }
 
 func (s *basicNFT) isApprovedOrOwner(addr std.Address, tid TokenID) bool {
-	owner, found := s.owners.Get(string(tid))
+	owner, found := s.owners.Get(tid.String())
 	if !found {
 		return false
 	}
 
-	if addr == owner.(std.Address) || s.IsApprovedForAll(owner.(std.Address), addr) {
+	ownerAddr, ok := owner.(std.Address)
+	if !ok {
+		return false
+	}
```

Comment:
Code could be simpler without the type cast check.
Right now owner addresses are validated before being setted, so all addresses should be valid.